### PR TITLE
[dynamo][reland] API Support for nn.Module

### PR DESCRIPTION
**Pull request by anijain2305**
Fixes #ISSUE_NUMBER

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/eval_frame.py [hunk 1]
```diff
@@ -44,6 +45,27 @@
...
+        self._orig_mod = mod
+        self.dynamo_ctx = dynamo_ctx
+
+    def __getattr__(self, name):
+        if name == "_orig_mod":
+            return self._modules["_orig_mod"]
+        return getattr(self._orig_mod, name)
+
+    def forward(self, *args, **kwargs):
+        return self.dynamo_ctx(self._orig_mod.forward)(*args, **kwargs)
```
Comment (anijain2305) [inline torch/_dynamo/eval_frame.py:66]: This is the main delta from last time. In the last PR, we were monkey patching. In this one, we are directly setting up the forward function here.

**Review by ezyang — APPROVED** [review]
I agree the delta should work

**Pull request closed without merging**